packages/app/pages/Api/agent.tsx

```diff
@@ -0,0 +1,145 @@
+      <div>name: {props.agent.name}</div>
+      <div>type: {props.agent.type}</div>
+      <div>image: {props.agent.image}</div>
+      <div>note: {props.agent.note}</div>
+      <div>ownedEconomicResources: {concatArray(props.agent.ownedEconomicResources)}</div>
+      <div>agentProcesses: {concatArray(props.agent.agentProcesses)}</div>
+      <div>agentPlans: {concatArray(props.agent.agentPlans)}</div>
+      <div>agentEconomicEvents: {concatArray(props.agent.agentEconomicEvents)}</div>
+      <div>agentCommitments: {concatArray(props.agent.agentCommitments)}</div>
+      <div>agentRelationships: {concatArray(props.agent.agentRelationships)}</div>
+      <div>agentRoles: {concatArray(props.agent.agentRoles)}</div>
+      <div>agentRecipess: {concatArray(props.agent.agentRecipes)}</div>
+      <br/>
+    </div>
+  );
+};
+
+function concatArray(array: JSON[]) {
+  var retValue = "";
+  console.log("Array: ", array, "\nIndex 0: ", array[0]);
+  if (array[0] == null) {
+    return "none"
+  }
+  retValue = array[0].id;
+  for (var i = 1; i < array.length; i++) {
+    retValue += ", ";
+    retValue += array[i].id;
+  }
+  return retValue
+}
+
+const AgentField = (props) => {
+  return(
+    <div>
+      <form onSubmit={props.onSubmitAction}>
+        Enter an Id: <input type="text" name="value" onChange={props.setAgent}/>
+        <input type="submit" value="query"/>
+      </form>
+    </div>
+  );
+};
+
+export const GetMyAgent = getMyAgent(({ agent, loading, error}) => {
+  return (
+    loading ? <strong>Loading...</strong> : (
+      error ? <p style={{color: "#F00"}}>API error</p> : (
+        <div>
+          <div>
+            {<Agent agent={agent}/>}
+          </div>
+        </div>
+      )
+    )
+  );
+});
+
+export const GetAllAgents = getAllAgents(({ agent, loading, error}) => {
+  return (
+    loading ? <strong>Loading...</strong> : (
+      error ? <p style={{color: "#F00"}}>API error</p> : (
+        <div>
+          <div>
+            {concatArray(agent)}
+          </div>
+        </div>
+      )
+    )
+  );
+});
+
+export const GetSingleAgent = getAgentById(({ agent, loading, error }) => {
+    return (
+      loading ? <strong>Loading...</strong> : (
+        error ? <p style={{color: "#F00"}}>API error</p> : (
+          <div>
+            <div>
+              <Agent agent={agent}/>
+            </div>
+          </div>
+        )
+      )
+    );
+  }
+);
+
+class App extends React.Component {
+
+  state = {
+    getOneAgentId: null,
+    setOneAgentId: null
+  };
+
+  // Runs every time the input field changes
+  getAgentById = (event) => {
+    this.setState({setOneAgentId: parseInt(event.target.value, 10)});
+  };
+
+  // Runs when "submit" is selected
+  stopRefresh = (event) => {
+    // Sets the value to query to the current value of the input field
+    this.setState({getOneAgentId: this.state.setOneAgentId});
+    event.preventDefault();
+  };
+
+  render() {
+    const {getOneAgentId} = this.state;
```

Comment:
This would be much easier to read as
`const getOneAgentId = this.state.getOneAgentId;`
instead of formatting the state object into just the single component

Reply:
I did this so that if I had needed additional state variables, I could add them (i.e. const {getOneAgentId, x} = this.state) but given I only have one, I would consider changing it if you think it is cleaner. What do you think?

Reply:
For now, since this API component seems to be feature complete, I think it would be unlikely that more variables are going to go into the state. I (personally) would refactor it to be a single assignment and if more variables are needed from the state in the future pull it out like you have currently. The change is up to you, though. I still approve either way.

Reply:
I am going to keep it this way. if in the future anyone copy-pastes this, it will be better for them to have it this way. marking as resolved

Reply:
resolved